app/aem/core/src/main/kotlin/com/cognifide/apm/core/endpoints/ScriptUploadServlet.kt

```diff
@@ -60,7 +60,7 @@ class ScriptUploadServlet : AbstractFormServlet<ScriptUploadForm>(ScriptUploadFo
 
     override fun doPost(form: ScriptUploadForm, resourceResolver: ResourceResolver): ResponseEntity<Any> {
         return try {
-            val script = scriptStorage.save(form.fileName, form.file, form.toLaunchMetadata(), form.overwrite, resourceResolver)
+            val script = scriptStorage.save(form.savePath, form.fileName, form.file, form.toLaunchMetadata(), form.overwrite, resourceResolver)

```

Comment:
If you are already changing the interface here, I would try to minimize the number of arguments. 6 arguments in a method call are too much. In this case, you can pass the whole `form` as 5 out of 6 arguments comes from the `form`. However, this method comes from the interface so it may not be that easy to change it. If this method is used in different contexts where `form` is not available, you can try to use other techniques described here: https://refactoring.guru/pl/smells/long-parameter-list